tf2_pykdl_ros/src/tf2_pykdl_ros/tf2_pykdl_ros.py

```diff
@@ -0,0 +1,189 @@
+from typing import Union
+from geometry_msgs.msg import PointStamped, PoseStamped, TransformStamped
+import PyKDL as kdl
+from pykdl_ros import VectorStamped, FrameStamped
+import tf2_ros
+
+
+def transform_to_kdl(t):
+    """
+    Convert a geometry_msgs Transform message to a PyKDL Frame.
+
+    :param t: The Transform message to convert.
+    :type t: geometry_msgs.msg.TransformStamped
+    :return: The converted PyKDL frame.
+    :rtype: PyKDL.Frame
+    """
+
+    return kdl.Frame(
+        kdl.Rotation.Quaternion(
+            t.transform.rotation.x, t.transform.rotation.y, t.transform.rotation.z, t.transform.rotation.w
+        ),
+        kdl.Vector(t.transform.translation.x, t.transform.translation.y, t.transform.translation.z),
+    )
+
+
+def to_msg_vector(vector: VectorStamped) -> PointStamped:
+    """
+    Convert a VectorStamped to a geometry_msgs PointStamped message.
+
+    :param vector: The vector to convert.
+    :type vector: pykdl_ros.VectorStamped
+    :return: The converted vector/point.
+    :rtype: geometry_msgs.msg.PointStamped
+    """
+    msg = PointStamped()
+    msg.header = vector.header
+    msg.point.x = vector.vector[0]
+    msg.point.y = vector.vector[1]
+    msg.point.z = vector.vector[2]
+    return msg
+
+
+tf2_ros.ConvertRegistration().add_convert((VectorStamped, PointStamped), to_msg_vector)
+tf2_ros.ConvertRegistration().add_to_msg(VectorStamped, to_msg_vector)
+
+
+def from_msg_vector(msg: Union[PointStamped, PoseStamped]) -> VectorStamped:
+    """
+    Convert a PointStamped message to a stamped VectorStamped.
+
+    :param msg: The PointStamped/PoseStamped message to convert.
+    :type msg: Union[geometry_msgs.msg.PointStamped, geometry.msg.PoseStamped]
+    :return: The timestamped converted PyKDL vector.
+    :rtype: pykdl_ros.VectorStamped
+    """
+    if isinstance(msg, PointStamped):
+        v = msg.point
+    elif isinstance(msg, PoseStamped):
+        v = msg.pose.position
+    else:
+        raise TypeError(f"msg should be PointStamped or PoseStamped, not '{type(msg)}'")
+    vector = VectorStamped.from_xyz(v.x, v.y, v.z, msg.header.stamp, msg.header.frame_id)
+    return vector
+
+
+tf2_ros.ConvertRegistration().add_convert((PointStamped, VectorStamped), from_msg_vector)
+tf2_ros.ConvertRegistration().add_convert((PoseStamped, VectorStamped), from_msg_vector)
+tf2_ros.ConvertRegistration().add_from_msg(VectorStamped, from_msg_vector)
+
+
+def convert_vector(vector: VectorStamped):
+    """
+    Convert a stamped PyKDL Vector to a stamped PyKDL Vector.
+
+    :param vector: The vector to convert.
+    :type vector: pykdl_ros.VectorStamped
+    :return: The timestamped converted PyKDL vector.
+    :rtype: pykdl_ros.VectorStamped
+    """
+    return VectorStamped(kdl.Vector(vector.vector), vector.header.stamp, vector.header.frame_id)
```

Comment:
the naming 'vector.vector' is a little weird...

Reply:
Yes, but that is the way it is. That is also the case with Stamped messages

Reply:
I still don't get this. You convert a Stamped PyKDL vector to a stamped PyKDL vector. Isn't this just a copy?